compiler/noirc_evaluator/src/acir/acir_variable.rs

```diff
@@ -1084,11 +1084,22 @@ impl<F: AcirField, B: BlackBoxFunctionSolver<F>> AcirContext<F, B> {
         &mut self,
         lhs: AcirVar,
         rhs: AcirVar,
+        typ: AcirType,
         bit_size: u32,
         predicate: AcirVar,
     ) -> Result<AcirVar, RuntimeError> {
-        let (_, remainder) = self.euclidean_division_var(lhs, rhs, bit_size, predicate)?;
-        Ok(remainder)
+        let numeric_type = match typ {
+            AcirType::NumericType(numeric_type) => numeric_type,
+            AcirType::Array(_, _) => {
+                todo!("cannot modulo arrays. This should have been caught by the frontend")
```

Comment:
```suggestion
                unreachable!("cannot modulo arrays. This should have been caught by the frontend")
```

`todo` implies that we don't yet disallow this in the frontend, which I'd be very surprised by.

Reply:
Done